File: examples/python/streaming.py
"""Streaming chat completion via AccelMars Gateway.

Prerequisites:
    pip install openai
    gateway serve  (or GATEWAY_MODE=mock gateway serve)
"""
import os
from openai import OpenAI

GATEWAY_URL = os.environ.get("ACCELMARS_GATEWAY_URL", "http://localhost:4000")

client = OpenAI(
    base_url=f"{GATEWAY_URL}/v1",
    api_key="local",
)

stream = client.chat.completions.create(
    model="standard",
    messages=[{"role": "user", "content": "Count to 5, one number per line."}],
    stream=True,
)

for chunk in stream:
    delta = chunk.choices[0].delta.content
    if delta is not None:
        print(delta, end="", flush=True)

print()

File: examples/ts/streaming.ts
/**
 * Streaming chat completion via AccelMars Gateway.
 *
 * Prerequisites:
 *   npm install
 *   gateway serve  (or GATEWAY_MODE=mock gateway serve)
 */
import OpenAI from "openai";

const GATEWAY_URL = process.env.ACCELMARS_GATEWAY_URL ?? "http://localhost:4000";

const client = new OpenAI({
  baseURL: `${GATEWAY_URL}/v1`,
  apiKey: "local",
});

const stream = await client.chat.completions.create({
  model: "standard",
  messages: [{ role: "user", content: "Count to 5, one number per line." }],
  stream: true,
});

for await (const chunk of stream) {
  process.stdout.write(chunk.choices[0]?.delta?.content ?? "");
}

console.log();
